core/templates/dev/head/pages/exploration-editor-page/services/change-list.service.ts

```diff
@@ -28,56 +28,58 @@ angular.module('oppia').factory('ChangeListService', [
   function(
       $log, $rootScope, AlertsService, AutosaveInfoModalsService,
       ExplorationDataService) {
-    // TODO(sll): Implement undo, redo functionality. Show a message on each
-    // step saying what the step is doing.
-    // TODO(sll): Allow the user to view the list of changes made so far, as
-    // well as the list of changes in the undo stack.
+    var ChangeListService = function() {
+      // TODO(sll): Implement undo, redo functionality. Show a message on each
+      // step saying what the step is doing.
+      // TODO(sll): Allow the user to view the list of changes made so far, as
+      // well as the list of changes in the undo stack.
 
-    // Temporary buffer for changes made to the exploration.
-    var explorationChangeList = [];
-    // Stack for storing undone changes. The last element is the most recently
-    // undone change.
-    var undoneChangeStack = [];
+      // Temporary buffer for changes made to the exploration.
+      this.explorationChangeList = [];
+      // Stack for storing undone changes. The last element is the most recently
+      // undone change.
+      this.undoneChangeStack = [];
 
-    // All these constants should correspond to those in exp_domain.py.
-    // TODO(sll): Enforce this in code.
-    var CMD_ADD_STATE = 'add_state';
-    var CMD_RENAME_STATE = 'rename_state';
-    var CMD_DELETE_STATE = 'delete_state';
-    var CMD_EDIT_STATE_PROPERTY = 'edit_state_property';
-    var CMD_EDIT_EXPLORATION_PROPERTY = 'edit_exploration_property';
+      // All these constants should correspond to those in exp_domain.py.
+      // TODO(sll): Enforce this in code.
+      this.CMD_ADD_STATE = 'add_state';
+      this.CMD_RENAME_STATE = 'rename_state';
+      this.CMD_DELETE_STATE = 'delete_state';
+      this.CMD_EDIT_STATE_PROPERTY = 'edit_state_property';
+      this.CMD_EDIT_EXPLORATION_PROPERTY = 'edit_exploration_property';
 
-    var ALLOWED_EXPLORATION_BACKEND_NAMES = {
-      category: true,
-      init_state_name: true,
-      language_code: true,
-      objective: true,
-      param_changes: true,
-      param_specs: true,
-      tags: true,
-      title: true,
-      auto_tts_enabled: true,
-      correctness_feedback_enabled: true
-    };
+      this.ALLOWED_EXPLORATION_BACKEND_NAMES = {
```

Comment:
Since these are constants, let's leave them out of the domain objects and use plain-old `var`.